src/meet-calendar.ts

```diff
@@ -0,0 +1,688 @@
+        $("#xOnCal").length != 0 || // readonly location
+        $("#xDescIn").length != 0 || // editable description
+        $("#xDesc").length != 0) && // readonly description
+      !this.isButtonPresent()
+    ) {
+      this.updateMeetingId();
+      this.addJitsiButton();
+    }
+  }
+
+  /**
+   * The event location.
+   */
+  get location(): LocationWrapper {
+    if (!this.locationInstance) {
+      this.locationInstance = new G2Location();
+    }
+    return this.locationInstance;
+  }
+
+  /**
+   * The button container holding jitsi button.
+   */
+  get buttonContainer() {
+    // we will create a new raw to place the button
+    // this row will be before the notifications row
+    let neighbor = $("#xNtList").parent();
+    if (neighbor.length == 0) {
+      return null;
+    }
+
+    let buttonContainer = $("#jitsi_button_container");
+    if (buttonContainer.length !== 0) {
+      return buttonContainer.find("content");
+    }
+
+    const newRow = $('<div class = "FrSOzf" />');
+    newRow.append(`
+            <div class = "tzcF6">
+                <div class = "DPvwYc jitsi_edit_page_icon"/>
+            </div>
+        `);
+
+    newRow.append(`
+            <div class = "j3nyw">
+                <div class = "BY5aAd">
+                    <div role = "button"
+                        class = "uArJ5e UQuaGc Y5sE8d"
+                        id="jitsi_button_container">
+                        <content class = "CwaK9">
+                            <span id="jitsi_button"
+                                    class="RveJvd snByac">
+                            </span>
+                        </content>
+                    </div>
+                </div>
+            </div>
+        `);
+
+    newRow.insertBefore(neighbor);
+
+    return newRow.find("content");
+  }
+
+  /**
+   * Adds the jitsi button in buttonContainer.
+   */
+  addJitsiButton() {
+    var container = this.buttonContainer;
+    if (!container) return false;
+
+    this.description?.update(this.location);
+  }
+
+  /**
+   * The event description.
+   */
+  get description(): DescriptionWrapper {
+    if (!this.descriptionInstance)
+      this.descriptionInstance = new G2Description(this);
+    return this.descriptionInstance;
+  }
+}
+
+/**
+ * The google calendar specific implementation of the location field in the
+ * event page.
+ */
+class G2Location extends LocationWrapper {
+  private _getSelector() {
+    return $("#xLocIn input[jsname=YPqjbf][role=combobox]");
+  }
+
+  private _getLocationElement() {
+    let elem = this._getSelector();
+
+    if (elem.length === 0) {
+      // this is the case where location is not editable
+      let element = $("#xOnCal")[0];
+
+      if (!element) {
+        return undefined;
+      }
+
+      return {
+        val: () => element.innerHTML,
+      };
+    }
+
+    return elem;
+  }
+
+  /**
+   * The text from the location input field.
+   */
+  get text(): string | undefined {
```

Comment:
either this should be renamed to `html()` or https://github.com/brave/jidesha/pull/2/files#diff-0d87d0858bdb97f9e337a8816f11b4c52a42a106faf5c119c89834ba8c783358R414 should be using innerText

Reply:
code simplified to address this confusion